mmpose/datasets/datasets/gesture/nvgesture_dataset.py

```diff
@@ -0,0 +1,168 @@
+        if use_bbox:
+            with open(self.bbox_file, 'r') as f:
+                bboxes = json.load(f)
+
+        for sample in samples:
+            sample = sample.strip().split()
+            sample = {
+                item.split(':', 1)[0]: item.split(':', 1)[1]
+                for item in sample
+            }
+            path = sample['path'][2:]
+            for key in ('depth', 'color'):
+                fname, start, end = sample[key].split(':')
+                sample[key] = {
+                    'path': os.path.join(path, fname + '.avi'),
+                    'valid_frames': (eval(start), eval(end))
+                }
+            sample['flow'] = {
+                'path': sample['color']['path'].replace('color', 'flow'),
+                'valid_frames': sample['color']['valid_frames']
+            }
+            sample['rgb'] = sample['color']
+            sample['label'] = eval(sample['label']) - 1
+
+            if use_bbox:
+                sample['bbox'] = bboxes[path]
+
+            del sample['path'], sample['duo_left'], sample['color']
+            db.append(sample)
+
+        return db
+
+    def _get_single(self, idx):
+        """Get anno for a single video."""
+        anno = defaultdict(list)
+        sample = self.db[self.vid_ids[idx]]
+
+        anno['label'] = sample['label']
+        anno['modality'] = self.modality
+        if 'bbox' in sample:
+            anno['bbox'] = sample['bbox']
+
+        for modal in self.modality:
+            anno['video_file'].append(
+                os.path.join(self.vid_prefix, sample[modal]['path']))
+            anno['valid_frames'].append(sample[modal]['valid_frames'])
+
+        return anno
+
+    def evaluate(self, results, res_folder=None, metric='AP', **kwargs):
+        """Evaluate nvgesture recognition results. The gesture prediction
+        results will be saved in ``${res_folder}/result_accuracy.json``.
+
+        Note:
+            - batch_size: N
+            - heatmap length: L
+
+        Args:
+            results (dict): Testing results containing the following
+                items:
+                - logits (dict[str, torch.tensor[N,25,L]]): For each item, \
+                    the key represents the modality of input video, while \
+                    the value represents the prediction of gesture. Three \
+                    dimensions represent batch, category and temporal \
+                    length, respectively.
+                - label (np.ndarray[N]): [center[0], center[1], scale[0], \
+                    scale[1],area, score]
+            res_folder (str, optional): The folder to save the testing
+                results. If not specified, a temp folder will be created.
+                Default: None.
+            metric (str | list[str]): Metric to be performed.
+                Options: 'AP'.
+
+        Returns:
+            dict: Evaluation results for evaluation metric.
+        """
+        if metric != 'AP':
```

Comment:
The metric could be a list ['AP'], I will recommend the following way.
```
metrics = metric if isinstance(metric, list) else [metric]
allowed_metrics = ['AP']
for metric in metrics:
   if metric not in allowed_metrics:
        raise KeyError(f'metric {metric} is not supported')
```

Reply:
Please also double check AP or mAP?

Reply:
Here mAP represents the mean AP among different modalities. It can be used to select the best model. `AP_{modal}` represents the AP of a single modality.